### DISCUSSION TITLE
请问可以支持Always translate English吗？

### DISCUSSION THREAD

**Discussion by theowenyoung** [chunibyocola/sc-translator-crx#83] [Q&A]
刚发现这个插件，excatly是我想要的插件！非常感谢作者的付出！

但是我似乎没有找到always translate English 的办法，有办法吗？

**Comment by theowenyoung** [comment]
这个插件可以支持自动翻译： https://github.com/FilipePS/Traduzir-paginas-web

如果有帮助的话。

**Comment by chunibyocola** [comment]
感谢支持。

有关自动翻译方面的我在 #27 中有提到：

> “翻译动态生成的内容”功能在 v4.5.0 中已经支持，预计在下一个版本中实装“网页自动翻译”的功能。
>
> “网页自动翻译”功能将会以域名白名单的形式执行，不会是自动判断网页语言与目标语言不同就自动翻译的这种模式，并且必须要启用“翻译动态生成的内容”选项后才能使用。在目前的构思中，检测到用户启用“网页自动翻译”后会在控制栏中加入一个复选框，用来启用/取消对当前网站的自动翻译。
>
> 需要注意的是，“在部分网站中使用可能会导致无限翻译请求”的问题还是无法解决，所以我个人是建议不开启的（绝大部分网站是没问题的）。

根据语言进行判断是非常容易误判的，所以这个功能我不太倾向于实装，从而选择以域名白名单的形式来代替。

**Comment by theowenyoung** [comment]
谢谢回复！ 我也想知道这个插件 https://github.com/FilipePS/Traduzir-paginas-web 是怎么避免在打开alwasy翻译之后的频繁请求问题的。

因为它支持自动翻译，这个功能对我还挺重要的，所以我尝试改了下它的代码，让它支持双语显示了，效果见： <https://twitter.com/OwenYoungZh/status/1588792579596111872> ，还没测试太多，但是我感觉已经满足我的需求了。

我fork的代码在这里： <https://github.com/theowenyoung/Traduzir-paginas-web>

**Reply by chunibyocola** [reply]
> 我也想知道这个插件 https://github.com/FilipePS/Traduzir-paginas-web 是怎么避免在打开alwasy翻译之后的频繁请求问题的。

这个插件有一个缓存的功能，但是这种缓存实际上用处并不大，因为只要一个字符不同，甚至 HTML 结构不同就会缓存失效。最后就会导致只有一些单词的缓存会被重复利用，造成了比较大的空间浪费。所以我觉得这个插件并没有解决这个问题，这个问题也几乎不能被解决。

**Reply by theowenyoung** [reply]
我之前没试过调用谷歌的翻译接口，这个接口是调多了就会被封吗？大概是调用多少次？有没有api key可以用避免这个问题？

**Reply by chunibyocola** [reply]
大概不会被封

**Reply by theowenyoung** [reply]
> 大概不会被封

那好像还行？不管怎么样我试用几天，我用的频率挺多的，要是有什么问题，我再在这里反馈。

**Reply by chunibyocola** [reply]
不需要反馈吧...因为新版本已经做好进入测试阶段，自动网页翻译已经是做好的了。

**Reply by theowenyoung** [reply]
是的，我clone下来试过最新的版本了，现在似乎只能手动添加域名支持自动翻译。我希望有always翻译英文的选项，然后自己对不需要的网站加黑名单。  我自己的用例是我经常浏览： https://www.buzzing.cc/ ，里面的网站太多了，不太适合一个一个的添加..
